vktrace/vktrace_common/vktrace_trace_packet_identifiers.h

```diff
@@ -71,33 +76,51 @@ typedef enum _VKTRACE_TRACE_PACKET_ID {
     VKTRACE_TPI_MARKER_API_GROUP_BEGIN,
     VKTRACE_TPI_MARKER_API_GROUP_END,
     VKTRACE_TPI_MARKER_TERMINATE_PROCESS,
+    VKTRACE_TPI_PORTABILITY_TABLE,
     VKTRACE_TPI_BEGIN_API_HERE  // this enum should always be the last in the list. Feel free to insert new ID above this one.
 } VKTRACE_TRACE_PACKET_ID;
 
+#define VKTRACE_BIG_ENDIAN 0
+#define VKTRACE_LITTLE_ENDIAN 1
+
 typedef struct {
     uint8_t id;
     uint8_t is_64_bit;
 } vktrace_tracer_info;
 
-#if defined(__LP64__) || defined(_WIN64) || (defined(__x86_64__) && !defined(__ILP32__)) || defined(_M_X64) || defined(__ia64) || \
-    defined(_M_IA64) || defined(__aarch64__) || defined(__powerpc64__)
+// vktrace_trace_file_header and vktrace_trace_packet_header are written and read
+// by both host and target CPU, which can be different architectures.
+// All fields in these structs are 64-bit aligned.
 
 typedef struct {
     uint16_t trace_file_version;
+    uint16_t reserved1;
+    uint16_t reserved2;
+    uint16_t reserved3;
+    uint64_t magic;
     uint32_t uuid[4];
     uint64_t first_packet_offset;  // will be size of header including size of tracer_id_array and state_snapshot_path/binary
-    uint8_t tracer_count;          // number of tracers referenced in this trace file
+    uint64_t tracer_count;         // number of tracers referenced in this trace file
     vktrace_tracer_info
         tracer_id_array[VKTRACE_MAX_TRACER_ID_ARRAY_SIZE];  // array of tracer_ids and values which are referenced in the trace file
     uint64_t trace_start_time;
+    uint64_t portability_table_valid;
+
+    // Description of platform on which this trace was created
+    uint64_t endianess;
+    uint64_t ptrsize;
+    uint64_t arch;
+    uint64_t os;
+    uint64_t gpu;
+    uint64_t drv_vers;
 } vktrace_trace_file_header;
 
 typedef struct {
     uint64_t size;  // total size, including extra data, needed to get to the next packet_header
     uint64_t global_packet_index;
-    uint8_t tracer_id;   // TODO: need to uniquely identify tracers in a way that is known by the replayer
-    uint16_t packet_id;  // VKTRACE_TRACE_PACKET_ID (or one of the api-specific IDs)
-    uint32_t thread_id;
+    uint64_t tracer_id;  // TODO: need to uniquely identify tracers in a way that is known by the replayer
+    uint64_t packet_id;  // VKTRACE_TRACE_PACKET_ID (or one of the api-specific IDs)
+    uint64_t thread_id;
```

Comment:
Does this change break the backward compatibility? The trace file captured with the old tracer may not be retraced with the new vkreplayer.